src/subdomains/supporting/bank-tx/bank-tx/services/bank-tx-batch.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@nestjs/common';
+import { BankTxBatch } from '../entities/bank-tx-batch.entity';
+import { BankTxBatchRepository } from '../repositories/bank-tx-batch.repository';
+
+@Injectable()
+export class BankTxBatchService {
+  constructor(private readonly bankTxBatchRepo: BankTxBatchRepository) {}
+
+  async getBankTxBatchByIban(iban: string): Promise<BankTxBatch> {
+    return this.bankTxBatchRepo.findOneBy({ iban });
+  }
```

Comment:
We should probably get the latest?

Reply:
Ah dump, I thought about it before I pushed it but forgot it 😅